packages/beacon-node/src/chain/chain.ts

```diff
@@ -222,13 +222,13 @@ export class BeaconChain implements IBeaconChain {
         ? anchorState
         : createCachedBeaconState(anchorState, {
             config,
-            pubkey2index: new PubkeyIndexMap(),
-            index2pubkey: [],
+            finalizedPubkey2index: new PubkeyIndexMap(),
```

Comment:
we don't need to change names, just add jsdoc to the `pubkey2index` and `index2pubkey` that they track finalized values,

also note that here they will get populated in createCachedBeaconState but thats ok since this `anchorState` is finalized (may be ass this as comment here for clarity)